go/obscuronode/enclave/state.go

```diff
@@ -271,9 +278,9 @@ func processDeposits(fromBlock *types.Block, toBlock *types.Block, blockResolver
 			break
 		}
 		for _, tx := range b.Transactions() {
-			t := obscurocommon.TxData(tx)
+			t := txHandler.UnPackTx(tx)
 			// transactions to a hardcoded bridge address
-			if t.TxType == obscurocommon.DepositTx {
+			if t != nil && t.TxType == obscurocommon.DepositTx {
```

Comment:
when can the unpacked tx be null?
is it an error, expected?

Reply:
Unpacked txs can be null when they are not obscuro-related transactions. So we just ignore them.

That was my line of thought. If you Unpack an Eth transactions it will only return obscuro transactions (or nil). If that makes sense ?

Reply:
Resolved in chat 👍

Reply:
I thought we would log when `t==nil`

Reply:
Yep! Logging is happening in https://github.com/obscuronet/obscuro-playground/blob/d11756ed32355d3da9cbffdedff3fd19ad4edc95/go/ethclient/mgmtcontractlib/handler.go#L103-L102